Assets/_Laz/Scripts/Environment/ElectricGate.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Laz
+{
+    [RequireComponent(typeof(BoxCollider))]
+    public class ElectricGate : MonoBehaviour
+    {
+
+        [SerializeField, Range(0f, 50f)] float _gateWidth = 10f;
+        [SerializeField] Transform _leftPost, _rightPost, _gate = default;
+        
+
+        private BoxCollider _gateCollider = null;
+
+        private void Awake()
+        {
+            _gateCollider = GetComponent<BoxCollider>();
+            ScaleGate();
+        }
+
+        private void OnCollisionEnter(Collision other)
+        {
+            if (other.collider.CompareTag(Tags.LazPlayer))
+            {
+                LazCoordinatorBehaviour lazCoordinator = other.gameObject.GetComponent<LazCoordinatorBehaviour>();
+
+                if (lazCoordinator != null)
+                {
+                    lazCoordinator.KillLaz();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Resizes gate according to gate width, and repositions posts
+        /// </summary>
+        private void ScaleGate()
```

Comment:
Don't leave comments(for 99% of the code), Comments lie the moment someone else changes this code, since no one normally changes comments in fear of messing something up.

The code should be self documenting. 

So what you could do here is leave the `ScaleGate()` as it is, but for each "if statement", it could be its own methods so it becomes more readable. 

so the Posts part, you could create separate methods for each post, e.g. `setPostPosition()`

Reply:
For sure. This isn't relevant to this code, but should I also be following that for public methods? Or those should all have xml comments?

Reply:
None of them should have comments. Only time that should have XML comments should be if your making api/sdk for outside developers

Reply:
So just incase my claims of no comments seems baseless. 

Clean Code: A Handbook of Agile Software Craftsmanship
by Amazon.ca
Learn more: https://www.amazon.ca/dp/0132350882/ref=cm_sw_em_r_mt_dp_3M8D8KCSBD346CP75XHP

Learnt Good Clean Coding habbits from here. I only recently read it, it's a great read. LMK if you're interested in reading it.

Reply:
Thats interesting, not sure I see the point of no xml comments whatsoever though. That seems like it makes IDE features useless haha

Reply:
Not everyone has the same IDE, and not everyone reads those comments, and those comments could be lying. 
So in general, at least from what I hear from other developers in just tech industry, "comments lie"
https://medium.com/young-coder/comments-are-useful-lies-f98c1c74476f

can read about it.